### PR TITLE
Adding fireflies

### PR DESCRIPTION
**Pull request by Venera3**
#### Summary

SUMMARY: Content "Add fireflies"

#### Purpose of change

Adding fireflies, passive luminous critters to pretty up the swamps, as part of #45030.

#### Describe the solution

Added fireflies, a nonhostile, almost blind critter to liven up the swamps a bit at night. They spawn in packs of 1-4 with a rareish frequency (for my reworked swamp group, see #44997 ). They lay eggs on a 15-day egg timer in the spring and summer and drop eggs on butchery. They also have their own harvest list and butchery snippets. 

#### Describe alternatives you've considered

Adding a flashbang spell for when the player bothers them, but since onhit/ondeath spell casting is not (yet) implemented that would have needed them to be hostile, and I wanted them to be some incidental pretty wildlife instead of Angry Bug Number 12. 

Not having them reproduce spontaneously, but that is the direction I want to take all bugs, and they will be hunted by a lot of things.

Having them drop a single egg upon death to keep the population stable, but game time catches up with the rot timer of monster drops, so each dead firefly generated a new one from day 5 onward.

#### Testing

They spawn, have the added butchery results and snippets, lay eggs and those rot into fireflies.

#### Additional context

Note, this uses some snippets from PR #44779 , but that throws no JSON errors other than the missing snippet on butchery. I can't really gauge the performance impact of these more than "it didn't make my shitty laptop catch fire", so let me know if they are too expensive and I should tone something down. From a balance standpoint their only use I could think of is as a live-in lightbulb, but that takes a lot of work for something trivial and I like the thought of a survivor reading a lab manual in firefly light.
Fireflies

### CODE REVIEW

File: data/json/items/comestibles/egg.json [hunk 1]
```diff
@@ -197,6 +197,18 @@
...
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "egg_firefly",
+    "name": "firefly egg",
+    "copy-from": "spider_egg",
+    "color": "yellow",
+    "symbol": "o",
+    "spoils_in": "1 day",
+    "description": "The fist-sized egg of a mutant firefly. Glows faintly in the dark.",
```
Comment (Maleclypse) [inline data/json/items/comestibles/egg.json:208]:
```suggestion
    "description": "The fist-sized egg of a mutant firefly.  Glows faintly in the dark.",
```

**Comment by dissociativity** [comment]
might be worth adding the eggs as an insect mutagen ingredient option?

**Comment by Venera3** [comment]
After I'm done with bugs, we might want to remove the swamp shack from the safe locations list if you know what I mean, but until then it is indeed quite cozy.

Redoing the mutagens to use only eggs is on the agenda, after I added some more critters that have them.

File: data/json/monsters/insect_spider.json [hunk 2]
```diff
@@ -518,6 +518,30 @@
     ],
     "extend": { "flags": [ "DESTROYS", "PUSH_MON", "PUSH_VEH" ] }
   },
+  {
+    "id": "mon_firefly",
+    "type": "MONSTER",
+    "name": { "str": "firefly", "str_pl": "fireflies" },
+    "description": "A mutated firefly grown to the size of a dog, brightly glowing abdomen bobbing up and down as it tries to impress its unseen mate.  It looks strangely uninterested in you and everything else.",
```
Comment (I-am-Erk) [inline data/json/monsters/insect_spider.json:525]:
```suggestion
    "description": "A mutated firefly grown to the size of a large housecat, brightly glowing abdomen bobbing up and down as it tries to impress its unseen mate.  It looks strangely uninterested in you and everything else.",
```
When I hear dog, I think a 15 kg hound more than something the size of a milk jug. Housecats are more consistent in size.

**Review by I-am-Erk — APPROVED** [review]
Needs conflicts resolved but otherwise I love it.

I wonder what weird things we could synthesize or derive from liters of firefly glow-goo.

**Comment by Venera3** [comment]
I thought about that, too. The most useful I managed to come up with was a makeshift glowstick from the goo and H2O2, but that was a lot of lines to solve a problem the player doesn't really have.

**Pull request merged**